src/x86_64/service.c

```diff
@@ -233,13 +225,24 @@ static boolean try_hw_seed(u64 * seed, boolean rdseed)
 u64 random_seed(void)
 {
     u64 seed = 0;
-    if (try_hw_seed(&seed, true))
+    if (have_rdseed && hw_seed(&seed, true))
         return seed;
-    if (try_hw_seed(&seed, false))
+    if (have_rdrand && hw_seed(&seed, false))
         return seed;
     return (u64)now();
 }
 
+static void init_hwrand(void)
```

Comment:
- We have U64_FROM_BIT() that you could use for this
- have_rdseed / have_rdrand should either be initialized to false when declared or set to false based on the cpuid calls here

Reply:
Results from cpuid() are u32s (eax, ebx etc). Updated to explicitly initialize booleans (though all of bss is zeroed anyway).